tenseal/cpp/tensors/tensor_storage.h

```diff
@@ -152,9 +138,6 @@ class TensorStorage {
     }
 
     TensorStorage<dtype_t>& broadcast_inplace(const vector<size_t>& new_shape) {
-        if (!can_broadcast(this->shape(), new_shape))
-            throw invalid_argument("invalid broadcast input");
-
         this->_data = xt::broadcast(this->_data, new_shape);
```

Comment:
Doesn't this throw an error if it's not possible to broadcast? I previously meant that we should catch the error at this point and provide information that the broadcast failed, or is the thrown error already sufficient?

Reply:
Some tests check if the broadcast throws an exception on incompatible dimensions, and they pass without that explicit check. Like https://github.com/OpenMined/TenSEAL/blob/xtensor/tests/cpp/tensors/plaintensor_test.cpp#L108